bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -0,0 +1,217 @@
+                ApiResponse apiResponse = gson.fromJson(response, ApiResponse.class);
+                updateChannels(apiResponse);
+            } catch (IOException e) {
+                logger.warn("Error opening connection to Meteo Alerte webservice : {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        } catch (MalformedURLException e) {
+            logger.warn("Malformed URL in Météo Alerte request : {}", queryUrl);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    /**
+     * Update the channel from the last Meteo Alerte data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannels(ApiResponse apiResponse) {
+        Arrays.stream(apiResponse.getRecords()).findFirst().ifPresent(record -> {
+            record.getFields().ifPresent(fields -> {
+                updateAlertString(WIND, fields.getVent());
+                updateAlertString(RAIN, fields.getPluieInondation());
+                updateAlertString(STORM, fields.getOrage());
+                updateAlertString(FLOOD, fields.getInondation());
+                updateAlertString(SNOW, fields.getNeige());
+                updateAlertString(HEAT, fields.getCanicule());
+                updateAlertString(FREEZE, fields.getGrandFroid());
+                updateAlertString(AVALANCHE, fields.getAvalanches());
+
+                fields.getDateInsert().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+                updateState(COMMENT, new StringType(fields.getVigilanceComment()));
+                updateIcon(WIND, fields.getVent());
+                updateIcon(RAIN, fields.getPluieInondation());
+                updateIcon(STORM, fields.getOrage());
+                updateIcon(FLOOD, fields.getInondation());
+                updateIcon(SNOW, fields.getNeige());
+                updateIcon(HEAT, fields.getCanicule());
+                updateIcon(FREEZE, fields.getGrandFroid());
+                updateIcon(AVALANCHE, fields.getAvalanches());
+            });
+        });
+    }
+
+    public void updateIcon(String channelId, String value) {
+        String iconChannelId = channelId + "-icon";
+        if (isLinked(iconChannelId)) {
+            String pictoName = channelId + (!value.isEmpty() ? "_" + value.toLowerCase() : "");
+            byte[] image = getImage("picto" + File.separator + pictoName + ".gif");
+            if (image != null) {
+                RawType picto = new RawType(image, "image/gif");
+                updateState(iconChannelId, picto);
+            }
+        }
+    }
+
+    private byte @Nullable [] getImage(String iconPath) {
+        URL url = FrameworkUtil.getBundle(getClass()).getResource(iconPath);
+        logger.debug("Path to icon image resource is: {}", url);
+        try {
+            InputStream in = url.openStream();
```

Comment:
Use try-with-resources to make sure that the url stream gets closed. You should also use a buffered input stream so that it will do batch reads behind the scenes for you which improves performance.
```suggestion
        try (InputStream in = new BufferedInputStream(url.openStream())){
```